File: solutions/cpp/space-age/1/space_age.cpp
#include "space_age.h"

namespace space_age {

    struct age {
        float seconds;
        float on_earth() {return seconds/EARTH_YEAR;};
        float on_mercury() {return on_earth()/0.2408467;};
        float on_venus() {return on_earth()/0.61519726;};
        float on_mars() {return on_earth()/1.8808158;};
        float on_jupiter() {return on_earth()/11.862615;};
        float on_saturn() {return on_earth()/29.447498;};
        float on_uranus() {return on_earth()/84.016846;};
        float on_neptune() {return on_earth()/164.79132;};
    };


}  // namespace space_age

File: solutions/python/space-age/1/space_age.py
class SpaceAge(object):
    def __init__(self, seconds):
        self.seconds = seconds
        self.earth_year = 31557600 

    def on_mercury(self):
        return round((self.seconds/self.earth_year) / 0.2408467,2)

    def on_venus(self):
        return round((self.seconds/self.earth_year) / 0.61519726,2)

    def on_earth(self):
        return round(self.seconds/self.earth_year,2)

    def on_mars(self):
        return round((self.seconds/self.earth_year) / 1.880815,2)

    def on_jupiter(self):
        return round((self.seconds/self.earth_year) / 11.86261,2)

    def on_saturn(self):
        return round((self.seconds/self.earth_year) / 29.44749,2)

    def on_uranus(self):
        return round((self.seconds/self.earth_year) / 84.016846,2)

    def on_neptune(self):
        return round((self.seconds/self.earth_year) / 164.7913,2)

